src/Shouldly/ShouldlyExtensionMethods/ShouldBe/ObjectGraphTestExtensions.cs

```diff
@@ -147,10 +147,17 @@ public static partial class ObjectGraphTestExtensions
         {
             foreach (var property in properties)
             {
+                if (property.GetIndexParameters().Length != 0)
+                {
+                    // There's no sensible way to compare indexers, as there does not exist a way to obtain a collection
+                    // of all values in a way that's common to all indexer implementations.
+                    throw new NotSupportedException("Comparing types that have indexers is not supported");
```

Comment:
Nit: could you add a period to make this user-facing message a complete sentence?